src/ip.rs

```diff
@@ -30,178 +27,235 @@ pub struct Ipv4Hdr {
 impl Ipv4Hdr {
     pub const LEN: usize = mem::size_of::<Ipv4Hdr>();
 
+    /// Returns the IP version field (should be 4).
+    #[inline]
+    pub fn version(&self) -> u8 {
+        (self.vihl >> 4) & 0xF
+    }
+
+    /// Returns the IP header length in bytes.
     #[inline]
     pub fn ihl(&self) -> u8 {
-        unsafe { mem::transmute(self._bitfield_1.get(0usize, 4u8) as u8) }
+        (self.vihl & 0xF) * 4
```

Comment:
We could consistently byte shift here as well:

```suggestion
        (self.vihl & 0xF) << 2
```

Reply:
I like it. I'll apply that change.